test/e2e/aadpodidentity_test.go

```diff
@@ -133,6 +134,21 @@ var _ = Describe("Kubernetes cluster using aad-pod-identity", func() {
 		deleteAllIdentityValidator()
 	})
 
+	It("should not delete the Immutable Identity from vmss when the deployment is deleted", func() {
+		setUpIdentityAndDeployment(immutableIdentity, "", "1")
```

Comment:
Ideally the e2e could:
1. Assign the immutable identity explicitly to the vmss/vm
2. Create an azure identity with the same immutable identity 
3. Run the identity validator check(ensure that a cycle has completed)
4. Now the identity validator is deleted (so it goes through a cycle of deletion) .
5. Check that count of  azureassignedidentity is 0 (confirming that the reconcile cycle is complete)
6. Check that the immutable identity is still present on the vm/vmss.